tests/test_worker_command_channel.py

```diff
@@ -0,0 +1,77 @@
+from file_writer_control.WorkerCommandChannel import WorkerCommandChannel
+from unittest.mock import Mock, patch
+from file_writer_control.WorkerStatus import WorkerStatus, WorkerState
+from file_writer_control.WriteJob import WriteJob
+from datetime import datetime, timedelta
+import time
+
+
+def get_test_job():
+    structure = "some structure"
+    file_name = "some file name"
+    start_time = datetime.now()
+    stop_time = start_time + timedelta(seconds=10)
+    service_id = "some service id"
+    broker = "some broker"
+    instrument_name = "some instrument name"
+    metadata = "some meta data"
+    run_name = "some run name"
+    return WriteJob(
+        nexus_structure=structure,
+        file_name=file_name,
+        broker=broker,
+        start_time=start_time,
+        stop_time=stop_time,
+        instrument_name=instrument_name,
+        run_name=run_name,
+        metadata=metadata,
+    )
+
+@patch("file_writer_control.WorkerCommandChannel.super")
+def test_list_idle_workers(TestClass):
+    under_test = WorkerCommandChannel("localhost:9090/hello")
+    all_workers = [WorkerStatus("id1"), WorkerStatus("id2"), WorkerStatus("id3"), WorkerStatus("id4")]
+    all_workers[0].state = WorkerState.WRITING
+    all_workers[1].state = WorkerState.UNAVAILABLE
+    all_workers[2].state = WorkerState.UNKNOWN
+    all_workers[3].state = WorkerState.IDLE
+    under_test.list_known_workers = Mock(return_value=all_workers)
+    idle_workers = under_test.get_idle_workers()
+    assert len(idle_workers) == 1
+    assert idle_workers[0].service_id == "id4"
+    assert idle_workers[0].state == WorkerState.IDLE
+
+
+@patch("file_writer_control.WorkerCommandChannel.super")
+def test_no_workers_within_5_seconds(TestClass):
+    under_test = WorkerCommandChannel("localhost:9090/hello")
+    under_test.get_idle_workers = Mock(return_value=[])
+    under_test.message_producer = Mock()
+    under_test.command_channel = Mock()
+    test_job = get_test_job()
+    under_test.try_start_job(test_job)
+    time.sleep(5)
```

Comment:
Does starting a job typically requires a lot of time?

Reply:
It can in some cases but that does not have anything to do with this `sleep`-call. `sleep` is called due to threading issues as we can not guarantee when a thread actually starts.

Reply:
5 seconds seems too magical to me. Why not 6 or 2? How do we decide this?

Reply:
It is a  long but not excessively so. With this value, I expect the unit test to pass at least 99% of the time on Jenkins.